inbm/dispatcher-agent/dispatcher/common/dispatcher_state.py

```diff
@@ -94,13 +115,13 @@ def write_dispatcher_state_to_state_file(state: DispatcherState) -> None:  # pra
     try:
         # if there's already a state file, read it in so we can update it and write back
         if is_dispatcher_state_file_exists():
-            with builtins.open(DISPATCHER_STATE_FILE, 'rb') as fd:
+            with builtins.open(NEW_DISPATCHER_STATE_FILE, 'rb') as fd:
                 # we are reading from a trusted state file here
                 val = pickle.load(fd)  # nosec B301
                 val.update(state)
                 state = val
             logger.debug(f"STATE WRITTEN: {state}")
-        with open(DISPATCHER_STATE_FILE, 'wb') as fd:
+        with open(NEW_DISPATCHER_STATE_FILE, 'wb') as fd:
```

Comment:
Maybe we could create a function that retrieves the path of the state file (either the new or old one) and use that path for the rest of the process.

Reply:
@yengliong93 I pushed a new commit -- how does this solution look? We want to try to migrate to the new state location on upgrade.

Reply:
Looks good to me.